wallet.md

```diff
@@ -0,0 +1,133 @@
+Wallet API
+==========
+
+BlockAdded Notifications
+------------------------
+```
+type NotifyBlockAddedRequestMessage struct {}
+
+type NotifyBlockAddedResponseMessage struct {
+	Error *RPCError
+}
+
+type BlockAddedNotificationMessage struct {
+	Block *RPCBlock
+}
+```
+ChainChanged Notifications
+--------------------------
+```
+type NotifyChainChangedRequestMessage struct {}
+
+type NotifyChainChangedResponseMessage strcut {
+	Error *RPCError
+}
+
+type ChainChangedNotificationMessage struct {
+	RemovedChainBlockHashes []string
+	AddedChainBlocks []*ChainBlock
+}
+
+type ChainBlock struct {
+	Hash string
+	AcceptedBlocks []*AcceptedBlock
+}
+
+type AcceptedBlock struct {
+	Hash string
+	AcceptedTransactionIDs []string
+}
+```
+UTXOsChanged Notifications
+--------------------------
+```
+type NotifyUTXOsChangedRequestMessage struct {
+	Addresses []string
+}
+
+type NotifyUTXOsChangedResponseMessage struct {
+	Error *RPCError
+}
+
+type UTXOsChangedNotificationMessage struct {
+	Entries []*UTXOsByAddressesEntry
+}
+```
+GetUTXOsByAddresses
+-------------------
+```
+type GetUTXOsByAddressesRequestMessage struct {
+	Addresses []string
+}
+
+type GetUTXOsByAddressesResponseMessage struct {
+	Entries []*UTXOsByAddressesEntry
+
+	Error *RPCError
+}
+
+type UTXOsByAddressesEntry struct {
+	Address string
+	Outpoint *RPCOutpoint
+	UTXOEntry *RPCUTXOEntry
+}
+```
+SubmitTransaction
+-----------------
+```
+type SubmitTransactionRequestMessage struct {
+	Transaction *RPCTransaction
+}
+```
+Dependant Types
+---------------
+```
+type RPCTransaction struct {
```

Comment:
Need to include ID (And if you are already there, I guess also Hash, although not really necassary).
Otherwise AcceptedBlock.AcceptedTransactionIDs makes no sense.